src/Semantic/REPL.hs

```diff
@@ -57,19 +51,15 @@ instance Exception Quit
 rubyREPL = repl (Proxy @'Language.Ruby) rubyParser
 
 repl proxy parser paths =
-  withOptions debugOptions $ \config logger statter ->
+  withOptions debugOptions $ \config _ _ ->
     runM
     . withDistribute
     . runCatch
-    . runResource
-    . withTimeout
     . runError @SomeException
-    . runTelemetryIgnoringStat (logOptionsFromConfig config)
-    . runTraceInTelemetry
-    . runReader (TaskSession config "-" False logger statter)
```

Comment:
Case in point, we are able to delete a bunch of stuff from `repl` now that parsing doesn’t require all of it.